lib/content-services/document-list/components/library-role-column/library-role-column.component.ts

```diff
@@ -15,44 +15,84 @@
  * limitations under the License.
  */
 
-import { Component, OnInit, Input } from '@angular/core';
+import {
+    Component,
+    OnInit,
+    Input,
+    ChangeDetectionStrategy,
+    ViewEncapsulation,
+    OnDestroy
+} from '@angular/core';
+import { Subscription, BehaviorSubject } from 'rxjs';
+import { AlfrescoApiService } from '@alfresco/adf-core';
+import { Node } from '@alfresco/js-api';
+import { ShareDataRow } from '../../data/share-data-row.model';
 
 @Component({
-  selector: 'adf-library-role-column',
-  template: `
-    <span title="{{ displayText | translate }}">
-      {{ displayText | translate }}
-    </span>
-  `,
-  host: { class: 'adf-library-role-column' }
+    selector: 'adf-library-role-column',
+    template: `
+        <span title="{{ (displayText$ | async) | translate }}">
+            {{ (displayText$ | async) | translate }}
+        </span>
+    `,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    host: { class: 'adf-library-role-column' }
 })
-export class LibraryRoleColumnComponent implements OnInit {
-  @Input()
-  context: any;
-
-  displayText: string;
-
-  ngOnInit() {
-    const node = this.context.row.node;
-    if (node && node.entry) {
-      const role: string = node.entry.role;
-      switch (role) {
-        case 'SiteManager':
-          this.displayText = 'LIBRARY.ROLE.MANAGER';
-          break;
-        case 'SiteCollaborator':
-          this.displayText = 'LIBRARY.ROLE.COLLABORATOR';
-          break;
-        case 'SiteContributor':
-          this.displayText = 'LIBRARY.ROLE.CONTRIBUTOR';
-          break;
-        case 'SiteConsumer':
-          this.displayText = 'LIBRARY.ROLE.CONSUMER';
-          break;
-        default:
-          this.displayText = '';
-          break;
-      }
+export class LibraryRoleColumnComponent implements OnInit, OnDestroy {
+    @Input()
+    context: any;
+
+    displayText$ = new BehaviorSubject<string>('');
+
+    private sub: Subscription;
+
+    constructor(private api: AlfrescoApiService) {}
+
+    ngOnInit() {
+        this.updateValue();
+
+        this.sub = this.api.nodeUpdated.subscribe((node: Node) => {
+            const row: ShareDataRow = this.context.row;
+            if (row) {
+                const { entry } = row.node;
+
+                if (entry === node) {
+                    row.node = { entry };
+                    this.updateValue();
+                }
+            }
+        });
+    }
+
+    protected updateValue() {
+        const node = this.context.row.node;
+        if (node && node.entry) {
+            const role: string = node.entry.role;
+            switch (role) {
+                case 'SiteManager':
```

Comment:
can we create an enum ?   we already have permissions.enum can we create permissions-site.enum ?

Reply:
Updated to use enum from js-api